shiny/test/_pytest.py

```diff
@@ -0,0 +1,34 @@
+from __future__ import annotations
+
+from pathlib import PurePath
+from typing import Generator
+
+import pytest
+
+from ._conftest import ShinyAppProc
+from .fixture import local_app_fixture_gen
+from .playwright import BrowserContext, Page
+
+
+# Make a single page fixture that can be used by all tests
+@pytest.fixture(scope="session")
+# By using a single page, the browser is only launched once and all tests run in the same tab / page.
+def session_page(browser: BrowserContext) -> Page:
+    return browser.new_page()
+
+
+@pytest.fixture(scope="function")
+# By going to `about:blank`, we _reset_ the page to a known state before each test.
+# It is not perfect, but it is faster than making a new page for each test.
+# This must be done before each test
+def page(session_page: Page) -> Page:
```

Comment:
Please add some docs for the public method

Reply:
Mention that this `page` is built on `session_page` which is maintained over the full session. What's new is `page` will visit `about:blank` to _reset_ between apps. 

Also mention that the default viewport size is set to `1920 x 1080` (1080p) for each test function.